flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/catalog/FunctionCatalogOperatorTable.java

```diff
@@ -74,7 +84,7 @@ public void lookupOperatorOverloads(
 			SqlSyntax syntax,
 			List<SqlOperator> operatorList,
 			SqlNameMatcher nameMatcher) {
-		if (!opName.isSimple()) {
+		if (opName.isStar()) {
```

Comment:
nit: `opName.isStar() && syntax == SqlSyntax.FUNCTION`.

I just add a test in local, `f0 is not null`  will also go into `lookupFunction` which should never happen. `FunctionCatalog.lookupFunction` should only handle functions, not prefix or postfix syntax.

Reply:
I think we should not limit to `SqlSyntax.FUNCTION`, there are other `SqlSyntax` too.
You can take a look to `CURRENT_TIME`, its syntax is `FUNCTION_ID`

Reply:
Then we should only allow `FUNCTION` and `FUNCTION_ID`?

Reply:
We don't know how much SqlSyntax...
If we pass on something we shouldn't, we can return to none. No problem.
But we didn't pass on what we should, which will lead to bugs.
So I think we can keep now.